### ISSUE TITLE
Octokit proxy cinfiguration

### ISSUE THREAD

**Issue by prime-optimus** [semantic-release/github#87]
I have an enterprise github setup which is behind a proxy. `Octokit` supports this configuration but there is no notion to pass proxy configuration in `@sementic-release/github` library.

Am I missing something?

I have tested this by changing the `Octokit` initialization in the following line (It works!):
https://github.com/semantic-release/github/blob/master/lib/get-client.js#L105

**Comment by pvdlg** [comment]
It seems the `proxy` option is deprecated in Octokit: https://github.com/aryannarora/rest.js/blob/master/test/integration/deprecations-test.js#L50.

I don't find any mention of a proxy config in the Readme nor in the API doc.
Looking at the code it seems Octokit uses [http-proxy-agent](https://www.npmjs.com/package/http-proxy-agent) and [node-https-proxy-agent](https://github.com/TooTallNate/node-https-proxy-agent) which accept the `http_proxy` environment variables as the proxy URL. Maybe you can try that?

@gr2m what was the `proxy` option replaced by? Do you have an example of how to set a proxy when instantiating Octokit?

**Comment by prime-optimus** [comment]
Yes, `proxy` is deprecated and is replaced by `agent` configuration. I used [https-proxy-agent](https://www.npmjs.com/package/https-proxy-agent) and it works fine if I change the code, [here](https://github.com/semantic-release/github/blob/master/lib/get-client.js#L105). 

What i really want is to expose proxies as a configuration, either accepting the configuration from upstream callers or as an environment variable so that `semantic release` can use that.

**Comment by pvdlg** [comment]
Can you share the code you changed to fix the issue? It's really difficult to see what you mean without the piece of code you are talking about.
Also, as mentioned before, I think there is already environment variable to configure the proxy URL: `http_proxy`.

**Comment by prime-optimus** [comment]
I just changed `Octokit` initialization [here](https://github.com/semantic-release/github/blob/master/lib/get-client.js#L105) to this:

```javascript
const github = new Octokit({baseUrl, agent: new HttpsProxyAgent("proxy-url")});
```

`Octokit` must be initialized with a proxy URL, I hard-coded it for testing, but I can get it from environment variable, `http_proxy` too.

**Comment by prime-optimus** [comment]
Thanks for the swift release guys. 👍 I just have one question, I think the choice of proxy environment variable must depend on `githubUrl`. Since the GitHub Url is secure(https) `https_proxy` environment variable must be picked.

Current changes would work fine if I put `https_proxy` variable in `http_proxy` variable too. But that would mess up other applications running on my environment.

What do you think about this?

**Comment by prime-optimus** [comment]
Thanks for the swift release guys! 👍

**Comment by pvdlg** [comment]
The fact that the target URL is `https` is independent of the protocol supported by the proxy. You can use an `http` proxy to reach an `https` URL and vice-versa.
The environment variable `HTTP_PROXY` is just a generic name, it doesn't imply that the proxy has to be contacted with `http` or `https` nor that the URL you target is `http` or `https`.

You should configure `HTTP_PROXY` to `https://your-proxy:3128` if it supports `https` and you want the communications between the plugin and the proxy to be secured.
If your proxy doesn't support `https` or you don't want to secure the communication between between the client and the proxy then you can configure `HTTP_PROXY` to `http://your-proxy:3128`.

**Comment by prime-optimus** [comment]
thanks, @pvdlg for the explanation.
